File: web/src/types/workflow.ts
export interface Workflow {
  id: number;
  name: string;
  description: string;
  project?: number | null;
  is_active: boolean;
  definition: Record<string, unknown>; // React Flow json
  created_at: string;
  updated_at: string;
}

export interface WorkflowExecution {
  id: number;
  workflow: number;
  status: 'QUEUED' | 'RUNNING' | 'SUCCESS' | 'FAILED' | 'CANCELLED';
  trigger_data: Record<string, unknown>;
  started_at: string;
  completed_at?: string | null;
  steps?: WorkflowExecutionStep[];
}

export interface WorkflowExecutionStep {
  id: number;
  execution: number;
  node_id: string;
  status: 'PENDING' | 'RUNNING' | 'SUCCESS' | 'FAILED';
  started_at: string;
  completed_at?: string | null;
  logs: string;
  error: string;
}

File: mobile/lib/models/workflow.dart
class Workflow {
  final int id;
  final String name;
  final String description;
  final bool isActive;
  final int? projectId;
  final DateTime createdAt;
  final DateTime updatedAt;

  Workflow({
    required this.id,
    required this.name,
    required this.description,
    required this.isActive,
    this.projectId,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Workflow.fromJson(Map<String, dynamic> json) {
    return Workflow(
      id: json['id'],
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      isActive: json['is_active'] ?? false,
      projectId: json['project'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }
}
